application/whatsapp/v0/send_message.go

```diff
@@ -0,0 +1,586 @@
+package whatsapp
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/instill-ai/component/base"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+// this file is used to handle 6 send tasks
+
+// Tasks:
+// 1. Send Text Message
+// 2. Send Media Message
+// 3. Send Location Message
+// 4. Send Contact Message
+// 5. Send Interactive CTA URL Button Message
+
+// types of messages: https://developers.facebook.com/docs/whatsapp/cloud-api/guides/send-messages
+
+// Objects (structs that are part of request struct; can be used in more than one task)
+
+type languageObject struct {
+	Code string `json:"code"`
+}
+
+type textObject struct {
+	Body       string `json:"body"`
+	PreviewURL bool   `json:"preview_url"`
+}
+
+type locationObject struct {
+	Latitude  string `json:"latitude"`
+	Longitude string `json:"longitude"`
+	Name      string `json:"name"`
+	Address   string `json:"address"`
+}
+
+type interactiveObject struct {
+	Type   string      `json:"type"`
+	Header interface{} `json:"header,omitempty"`
+	Body   struct {
+		Text string `json:"text"`
+	} `json:"body"`
+	Footer *footerObject `json:"footer,omitempty"`
+	Action actionObject  `json:"action"`
+}
+
+type footerObject struct {
+	Text string `json:"text"`
+}
+
+type actionObject struct {
+	Name      string `json:"name,omitempty"`
+	Parameter struct {
+		DisplayText string `json:"display_text"`
+		URL         string `json:"url"`
+	} `json:"parameters"` //I also don't know why the json is parameters instead of parameter, but the API works only when it is parameters
```

Comment:
Because the key is parameters
<img width="859" alt="image" src="https://github.com/user-attachments/assets/88211b80-fa38-4362-9979-e8214e2bc81f">

Reply:
And, please take out the comments that do not make document clear.